tests/e2e/flowtest/flowtest_helpers.py

```diff
@@ -0,0 +1,223 @@
+import logging
+
+from ocs_ci.ocs import node, platform_nodes, defaults, exceptions
+from ocs_ci.ocs.node import wait_for_nodes_status
+from ocs_ci.ocs.resources import pod as pod_helpers
+from ocs_ci.ocs.resources.pod import wait_for_storage_pods
+from ocs_ci.utility.utils import TimeoutSampler
+from tests.manage.mcg.helpers import s3_delete_object, s3_put_object, s3_get_object
+from tests.sanity_helpers import Sanity
+from ocs_ci.framework.testlib import ignore_leftovers
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources.pvc import delete_pvcs
+from tests import helpers
+from tests.manage.z_cluster.pvc_ops import create_pvcs
+
+logger = logging.getLogger(__name__)
+
+
+class FlowOperations:
+    def __init__(self):
+        """
+        Initialize Sanity instance
+
+        """
+        self.sanity_helpers = Sanity()
+
+    def node_operations_entry_criteria(self, node_type, number_of_nodes, network_fail_time=None):
+        """
+        Entry criteria function for node related operations
+        Args:
+            node_type (str): Tyoe of node
+            number_of_nodes (int): Number of nodes
+            network_fail_time (int): Total time to fail the network in a node
+        Returns:
+            tuple: containing the params used in Node operations
+
+        """
+        logger.info("Node operations entry: Verifying cluster health")
+        self.sanity_helpers.health_check(cluster_check=False)
+        logger.info("Node operations entry: Verifying StorageCluster pods are in running/completed state")
+        wait_for_storage_pods(timeout=300), 'Some pods were not in expected state'
+        logger.info("Node operations entry: Verifying whether node is ready")
+        wait_for_nodes_status(status=constants.NODE_READY, timeout=300)
+
+        logger.info("Getting parameters related to Node operation")
+        typed_nodes = node.get_typed_nodes(
+            node_type=node_type, num_of_nodes=number_of_nodes
+        )
+        factory = platform_nodes.PlatformNodesFactory()
+        nodes = factory.get_nodes_platform()
+
+        if network_fail_time:
+            return typed_nodes, nodes, network_fail_time
+        else:
+            return typed_nodes, nodes
+
+    def node_operations_exit_criteria(self):
+        """
+        Exit criteria verification function for node related operations
+
+        """
+        logger.info("Node operations exit: Verifying whether node is ready")
+        wait_for_nodes_status(status=constants.NODE_READY, timeout=300)
+        logger.info("Node operations exit: Verifying cluster health")
+        self.sanity_helpers.health_check(cluster_check=False, tries=80)
+        logger.info("Node operations exit: Verifying StorageCluster pods are in running/completed state")
+        wait_for_storage_pods(timeout=300), 'Some pods were not in expected state'
+
+        logging.info("Node operation: Exit criteria verification: Success")
+
+    def add_capacity_entry_criteria(self):
```

Comment:
I think we have something similar under tests.manage.z_cluster.cluster_expansion.test_add_capacity_entry_exit_criteria.TestAddCapacity.test_add_capacity

Reply:
True, but they are part of the test case itself and are scattered, this is a generic function.